views/default/object/widget/edit/num_display.php

```diff
@@ -22,15 +26,26 @@
 $vars['#label'] = $vars['label'];
 unset($vars['label']);
 
-if (!isset($vars['options'])) {
-	$vars['options'] = [5, 8, 10, 12, 15, 20];
-}
-
 $value = sanitize_int($widget->$name, false);
 if (!$value) {
-	$value = elgg_extract('default', $vars, $vars['options'][0]);
+	$value = (int) elgg_extract('default', $vars, 4);
 }
 $vars['value'] = $value;
-$vars['#type'] = 'select';
+
+$vars['step'] = (int) elgg_extract('step', $vars, 1);
+
+$min = (int) elgg_extract('min', $vars, 1);
```

Comment:
maybe sanitize `$min` to `max(1, $min);` to make sure it's not negative

Reply:
Actually, I had this check in at first and also ```$max = max($min, $max)``` but then I thought it might be too much. But I'll add it again.